### DISCUSSION TITLE
Truffle tests failling at differing points upon different runs

### DISCUSSION THREAD

**Discussion by msuscens** [ConsenSys-archive/ganache#3368] [Q&A]
Hi @davidmurdoch,

I think I have uncovered an issue, where truffle tests fail at differing points upon different executions (under ganache 7.3.2).  I've put together an example that shows the issue - see the Readme file that gives the results I'm getting.

I also tried executing the same truffle tests under ganache-cli v6.12.2.  The tests (in test case 02 file) typically all pass - but I do get an odd failure on some test runs.  However the much heavy tests where the Person token's breed() function is repeatedly called to create a family tree of People (tokens) never manages to complete the family tree setup - failing at different points part the way through this setup.

I use ganache-time-traveler in these test cases - so it may be related in some way to issue #3265

https://github.com/msuscens/People_Example

 % truffle version 
Truffle v5.5.3 (core: 5.5.3)
Ganache v7.0.1
Solidity - 0.8.13 (solc-js)
Node v14.15.5
Web3.js v1.5.3

Should I open this as an issue?

**Comment by MicaiahReid** [comment]
Hi @msuscens, thanks for opening this!

I made a local build of Ganache with the fix for #3265 and your tests are still failing, so sadly I don't think that PR will resolve this issue.

However, it's kind of hard to track down what the issue is here - the reproduction you've given us is a lot of code for us to walk through and understand in order to track down a potential issue in Ganache.

Can you open an issue if you get something specific as to what's going wrong with Ganache?

**Reply by msuscens** [reply]
Thanks @MicaiahReid for initially looking into this.  And yes, sorry it was a lot of code!!

As suggested I've written a somewhat simpler example and investigated further.  Please see the issue I've just raise: #3425
I'd greatly appreciate you taking a look :-)

Thanks again,
Mark
